.github/workflows/nightly.yaml

```diff
@@ -0,0 +1,156 @@
+---
+name: Nightly Builds
+
+on:
+  schedule:
+    cron: '0 0 * * *'  # Every night midnight
+
+  workflow_dispatch:   # Or manually - for testing
+
+jobs:
+  # Build the build-environment container, using it's workflow
+  build-env:
+    runs-on: ubuntu-latest
+    needs:
+      - set-git-refs
+
+    outputs:
+      date: ${{ steps.date.outputs.date }}
+
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Dockerhub Login
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Generate Date String
+        id: date
+        run: |
+          DATE="$(date +%Y-%m-%d)"
+          echo "date=${DATE}" >> $GITHUB_OUTPUT
+
+      - name: Build Buildenv Container
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/buildenv-radosgw:nightly-${{ steps.date.outputs.date }}
+          file: tools/build/Dockerfile.build-radosgw
```

Comment:
can't we rely on the containers we have on quay?

Reply:
Good point. Copied that code from the release pipeline with minimal modifications, so this is slipped past.